src/testing/euler_gaussian_bump.cpp

```diff
@@ -93,92 +73,48 @@ ::run_euler_gaussian_bump() const
-        // ode_solver->initialize_steady_polynomial_ramping (poly_degree);
+            param.flow_solver_param.poly_degree = solution_degree;
+            param.flow_solver_param.grid_degree = grid_degree;
+            param.flow_solver_param.num_refinements = igrid;
 
-        for (unsigned int igrid=0; igrid<n_grids; ++igrid) {
+            pcout << "\n" << "************************************" << "\n" << "POLYNOMIAL DEGREE " << solution_degree 
+                  << ", GRID NUMBER " << (igrid+1) << "/" << n_grids << "\n" << "************************************" << std::endl;
 
+            pcout << "\n" << "Creating FlowSolver" << std::endl;
 
-            //if (igrid!=0) {
-            //    dealii::LinearAlgebra::distributed::Vector<double> old_solution(dg->solution);
-            //    dealii::parallel::distributed::SolutionTransfer<dim, dealii::LinearAlgebra::distributed::Vector<double>, dealii::hp::DoFHandler<dim>> solution_transfer(dg->dof_handler);
-            //    solution_transfer.prepare_for_coarsening_and_refinement(old_solution);
-            //    dg->high_order_grid->prepare_for_coarsening_and_refinement();
-            //    grid->refine_global (1);
-            //    dg->high_order_grid->execute_coarsening_and_refinement(true);
-            //    dg->allocate_system ();
-            //    dg->solution.zero_out_ghosts();
-            //    solution_transfer.interpolate(dg->solution);
-            //    dg->solution.update_ghost_values();
-            //}
-
-            using Triangulation = dealii::parallel::distributed::Triangulation<dim>;
-            std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(
-                MPI_COMM_WORLD,
-                typename dealii::Triangulation<dim>::MeshSmoothing(
-                    dealii::Triangulation<dim>::smoothing_on_refinement |
-                    dealii::Triangulation<dim>::smoothing_on_coarsening));
-
-            const double channel_length = 3.0;
-            const double channel_height = 0.8;
-            Grids::gaussian_bump(*grid, n_subdivisions, channel_length, channel_height);
-            grid->refine_global(igrid);
+            std::unique_ptr<FlowSolver::FlowSolver<dim,nstate>> flow_solver = FlowSolver::FlowSolverFactory<dim,nstate>::select_flow_case(&param, parameter_handler);
 
-            const double solution_degree = poly_degree;
-            const double grid_degree = solution_degree+1;
-            std::shared_ptr < DGBase<dim, double> > dg = DGFactory<dim,double>::create_discontinuous_galerkin(&param, solution_degree, solution_degree, grid_degree, grid);
+            flow_solver->run();
 
-            // Initialize coarse grid solution with free-stream
-            dg->allocate_system ();
-            dealii::VectorTools::interpolate(dg->dof_handler, initial_conditions, dg->solution);
+            std::shared_ptr < DGBase<dim, double> > dg = flow_solver->dg;
```

Comment:
Why create a pointer to dg here? Why not just call flow_solver->dg when needed?

Reply:
it was done as a quick fix to avoid updating it everywhere below, is there any reason not to?

Reply:
fixed